CmsManager/CmsSnapshotManager.php

```diff
@@ -36,13 +35,20 @@ class CmsSnapshotManager extends BaseCmsPageManager
     protected $pages = array();
 
     /**
+     * @var string
+     */
+    protected $snapshotPageProxyClass;
+
+    /**
      * @param SnapshotManagerInterface $snapshotManager
      * @param TransformerInterface     $transformer
+     * @param string                   $snapshotPageProxyClass Namespace of SnaphostPageProxy class
      */
-    public function __construct(SnapshotManagerInterface $snapshotManager, TransformerInterface $transformer)
+    public function __construct(SnapshotManagerInterface $snapshotManager, TransformerInterface $transformer, $snapshotPageProxyClass = 'Sonata\PageBundle\Model\SnapshotPageProxy')
     {
         $this->snapshotManager = $snapshotManager;
         $this->transformer = $transformer;
+        $this->snapshotPageProxyClass = $snapshotPageProxyClass;
```

Comment:
Is this should be mandatory?

In this case, you should throw a deprecated error.